CoEpi/ble/Peripheral.swift

```diff
@@ -94,31 +95,35 @@ extension PeripheralImpl: CBPeripheralManagerDelegate {
     }
 
     func peripheralManager(_ peripheral: CBPeripheralManager, didReceiveRead request: CBATTRequest) {
-        os_log("Peripheral manager did receive read request: %@", request.description)
-
-        let identifier = UUID()
-        addNewContactEvent(with: identifier)
-        request.value = identifier.dataRepresentation
+        os_log("Peripheral manager did receive read request: %@", log: blePeripheralLog, request.description)
+        
+        let currentTimestamp = Int(Date().timeIntervalSince1970)
+        let currentCENKey = CENKey.generateAndStoreCENKey()
+        let CENData: Data = CEN.generateCENData(CENKey: currentCENKey.cenKey!, currentTs: currentTimestamp)
```

Comment:
I don't think timestamp needs to be an argument unless you're allowed to make CEN's at arbitrary timestamps. Could just get currentTs within the function.

Reply:
updated